src/azure-cli/azure/cli/command_modules/appconfig/_utils.py

```diff
@@ -34,17 +35,52 @@ def construct_connection_string(cmd, config_store_name):
+                    endpoint = store.endpoint
+                else:
+                    raise ValidationError('Multiple configuration stores found with name {}.'.format(config_store_name))
     except HttpResponseError as ex:
-        raise CLIError("Failed to get the list of App Configuration stores for the current user. Make sure that the account that logged in has sufficient permissions to access the App Configuration store.\n{}".format(str(ex)))
+        raise AzureResponseError("Failed to get the list of App Configuration stores for the current user. Make sure that the account that logged in has sufficient permissions to access the App Configuration store.\n{}".format(str(ex)))
+
+    if resource_group is not None and endpoint is not None:
+        return resource_group, endpoint
+
+    raise ResourceNotFoundError("Failed to find the App Configuration store '{}'.".format(config_store_name))
+
+
+def resolve_deleted_store_metadata(cmd, config_store_name, resource_group_name, location):
+    resource_group = None
+    metadata_location = None
+    try:
+        client = cf_configstore(cmd.cli_ctx)
+        deleted_stores = client.list_deleted()
+        for deleted_store in deleted_stores:
+            # configuration_store_id has a fixed structure /subscriptions/subscription_id/resourceGroups/resource_group_name/providers/Microsoft.AppConfiguration/configurationStores/configuration_store_name
+            metadata_resource_group = deleted_store.configuration_store_id.split('/')[4]
+            # match the name and additionally match resource group and location if available.
+            if deleted_store.name.lower() == config_store_name.lower() and (resource_group_name is None or resource_group_name.lower() == metadata_resource_group.lower()) and (location is None or location == deleted_store.location):
+                if metadata_location is None:
+                    resource_group = metadata_resource_group
+                    metadata_location = deleted_store.location
+                else:
+                    # It should reach here only when the user has provided only name. If they provide either location or resource group, we should be able to uniquely find the store.
+                    raise ValidationError('Multiple configuration stores found with name {}.'.format(config_store_name))
+    except HttpResponseError as ex:
+        raise AzureResponseError("Failed to get the list of deleted App Configuration stores for the current user. Make sure that the account that logged in has sufficient permissions to access the App Configuration store.\n{}".format(str(ex)))
+
+    if resource_group is not None and metadata_location is not None:
+        return resource_group, metadata_location
 
-    raise CLIError("Failed to find the App Configuration store '{}'.".format(config_store_name))
+    raise ResourceNotFoundError("Failed to find the deleted App Configuration store '{}'.".format(config_store_name))
```

Comment:
This error could be hit if the store name is correct, but user provided a location or RG that was incorrect. I feel that we should hint at this in the error message. How about this:
"Failed to find the deleted App Configuration store `storeName`. If you think that the store name is correct, please validate all your input parameters again."

Reply:
Great suggestion! Thank you!